client/components/DocumentList.js

```diff
@@ -0,0 +1,256 @@
+import React, { PropTypes } from 'react';
+import {Modal, Button, Row, Input, Pagination} from 'react-materialize';
+import moment from 'moment';
+import {Link} from 'react-router';
+import {connect} from 'react-redux';
+import DocumentTitle from '../components/DocumentListTitle';
+import DocumentContent from '../components/DocumentContent';
+import  * as DocumentAction from '../actions/DocumentActions';
+import Search from '../components/SearchDocument';
+import imagePath from '../img/cardReveal.jpg';
+import renderHTML from 'react-render-html';
+import TinyMCE from 'react-tinymce';
+import Prompt from '../components/Prompt';
+
+/**
+ * 
+ * 
+ * @class DocumentList
+ * @extends {React.Component}
+ */
+class DocumentList extends React.Component{
+  constructor (props) {
+    super(props);
+    const { updateDocument } = this.props;
+    const { deleteDocument } = this.props;
+    const { fetchDocuments } = this.props;
+    this.state = {
+    id: '',
+    title: '',
+    content: '',
+    access: 'public',
+    authorId: '',
+  };
+    this.onChange = this.onChange.bind(this);
+    this.onClick = this.onClick.bind(this);
+    this.handleEditorChange = this.handleEditorChange.bind(this);
+  }
+
+  /**
+   * 
+   * 
+   * @param {any} e 
+   * @returns 
+   * 
+   * @memberOf DocumentList
+   */
+  fieldChange(e) {
+      return this.setState({ [e.target.name]: e.target.value,  })
+    }
+
+  /**
+   * 
+   * 
+   * @param {any} event 
+   * 
+   * @memberOf DocumentList
+   */
+  onChange(event) {
+  this.setState({
+    [event.target.name]: event.target.value
+  });
+  } 
+
+  /**
+   * 
+   * 
+   * @param {any} e 
+   * 
+   * @memberOf DocumentList
+   */
+  handleEditorChange(e) {
+    this.setState({ content: e.target.getContent() });
+  }
+ 
+  /**
+   * 
+   * 
+   * @param {any} event 
+   * 
+   * @memberOf DocumentList
+   */
+  onClick(event) {
+  event.preventDefault();
+  this.props.deleteDocument(this.props.role.id).then(() => {
+    Materialize.toast('Document successfully deleted.', 5000);
```

Comment:
I like this better than the generous use of Materialize.toast in your actions which doesn't seem to be in the spirit of redux.